src/Dropdown.js

```diff
@@ -274,10 +274,10 @@ class Dropdown extends React.Component {
       labelledBy: id,
       bsClass: prefix(props, 'menu'),
       onClose: createChainedFunction(
-        child.props.onClose, onClose, this.handleClose,
+        child.props.onClose, onClose, (event) => this.handleClose(event, 'close'),
```

Comment:
`close` is a weird event type for a close action. `rootClose` might be nice, but it's only appropriate for the default implementation. i wonder if we should let the child deal with setting the event type here, something like

```
onRootClose = (e) => {
  this.props.onClose(e, { source: 'rootClose' });
}
```

in `<DropdownMenu>`.